apps/api/src/domain/models/resource.entity.ts

```diff
@@ -42,6 +54,25 @@ export abstract class Resource extends BaseDomainModel implements HasAggregateId
         id: this.id,
     });
 
+    /**
+     * The name of this method is a bit misleading. It merely adds an event
+     * to the list of historical events without updating the state.
+     *
+     * At present, we are not doing
+     * event sourcing. Rather, the event is created after successfully mutating
+     * a model's state and immediately before persisting the result to the database.
+     * The event is at present simply a record of a command that has succeeded
+     * historically for troubleshooting or migrations (e.g. opt-in to additional
+     * raw data from import event).
+     */
+    applyEvent<T extends Resource = Resource>(this: T, event: BaseDomainModel) {
```

Comment:
I'm assuming you are avoiding `createEvent`?

Reply:
- registerSuccessfulEvent
- persistEvent
- addEventToHistory